### PR TITLE
Locking & Voting (Tokenomics)

### PR DESCRIPTION
**Pull request by evgenyboxer**
# Description

This is the base branch for the veBAL page + getVeBAL modal.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Visual context

Please provide any relevant visual context for UI changes or additions. This could be static screenshots or a loom screencast.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have requested at least 2 reviews (If the PR is significant enough, use best judgement here)
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] If package-lock.json has changes, it was intentional.
- [ ] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/composables/queries/useAllowancesQuery.ts [hunk 1]
```diff
@@ -49,11 +49,13 @@ export default function useAllowancesQuery(
...
   const queryFn = async () => {
     console.log('Fetching', tokenAddresses.value.length, 'allowances');
-    return await tokenService.allowances.get(
+    const allowances = await tokenService.allowances.get(
       account.value,
       contractAddresses.value,
       tokens.value
     );
+
+    return allowances;
```
Comment (garethfuller) [inline src/composables/queries/useAllowancesQuery.ts:58]: I think this was probably to log allowances right? Can we revert to original please.

File: src/plugins/router.ts [hunk 2]
```diff
@@ -51,12 +53,6 @@ const routes: RouteRecordRaw[] = [
     component: PoolWithdrawPage,
     meta: { layout: 'PoolTransferLayout' }
   },
-  {
-    path: '/pool/migrate/:from/:to',
-    name: 'migrate-pool',
-    component: MigratePoolPage,
-    meta: { layout: 'FocusedLayout' }
-  },
```
Comment (garethfuller) [inline src/plugins/router.ts:59]: Why has the migration path been deleted here?

**Review by garethfuller — COMMENTED** [review]
This is my first pass. Code looks great, really clean and well structured.

I talked with Pon/Fernando about the term 'Unlock', lets use the word 'Redeem' instead. Please replace all references to Unlock in component names, labels and routes.

**Comment by garethfuller** [comment]
@evgenyboxer when unlocking, the link back to the vebal page is broken.

**Comment by garethfuller** [comment]
@evgenyboxer when my lock expired, I was unable to extend the lock again, it just kept reverting the tx. I had to unlock and then relock.

**Comment by markusbkoch** [comment]
The veBAL dashboard tells me I have no BPT to lock on this address 0x67dc6f6ffa7De114c33144f3b1630Fd1Eecd27dD, even though I do
image

The lock UI does recognize them
image

The only relevant thing I can think of is that I just withdrew BPT from the veBAL contract, so maybe there's an unhandled edged case in the dashboard that causes it to show like this? cc @garethfuller

**Comment by timjrobinson** [comment]
@markusbkoch That is fixed, was just not showing the BPT value unless you had veBAL locked.

File: src/composables/useTime.ts [hunk 3]
```diff
@@ -24,6 +28,16 @@ export function toJsTimestamp(unixTimestamp: number): number {
   return unixTimestamp * oneSecondInMs;
 }
 
+export function getPreviousThursday(date: Date): Date {
+  const daysSinceThursday = (date.getDay() - 4) % 7;
```
Comment (TomAFrench) [inline src/composables/useTime.ts:32]:
```suggestion
  const daysSinceThursday = (date.getDay() - 4 + 7) % 7;
```

This doesn't work for Mon-Wed currently as it doesn't handle negatives correctly so we need to bump up by 7

**Review by TomAFrench — CHANGES_REQUESTED** [review]

File: src/composables/useTime.ts (hunk 3 above)
Comment (TomAFrench) [reply src/composables/useTime.ts:32]: Granted, I think this bug got added by copy-pasting my snippet.

**Comment by garethfuller** [comment]
> Is this working correctly? total at the bottom says 0.0091 vs actual says 0.0086 veBAL
> 
> Screen Shot 2022-03-20 at 4 18 24 PM

@zekraken-bot the Total voting escrow value in the form is what you get after you've locked. The value on the right is your current veBAL. If this is not clear enough, we may need to change labels. cc @pkattera

**Pull request merged**